src/server.py

```diff
@@ -0,0 +1,183 @@
+
+
+import socket
+import sys
+
+
+def server():
+    """Recieve a message from the client and sends a response back."""
+    while True:
+        server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP)
+        address = ("127.0.0.1", 7030)
+        server_socket.bind(address)
+        server_socket.listen(1)
+        conn, addr = server_socket.accept()
+        buffer_length = 8
+        req = buffer_request(buffer_length, conn)
+        print(req)
+        message = parse_request(req)
+        if sys.version_info[0] == 2:
+            conn.sendall(message)
+        else:
+            conn.sendall(message.encode('utf-8'))
+        conn.close()
+        server_socket.close()
+
+
+def buffer_request(buffer_length, conn):
+    """Return a decoded client request."""
+    req = u""
+    while True:
+        addition = conn.recv(buffer_length).decode('utf-8')
+        req += addition
+        if len(addition) < buffer_length:
+            break
+    return req.replace('\\r\\n', '\r\n')
+
+
+def parse_request(request):
+    """Return a server response based on client request formatting."""
+    if not method_validation(request):
+        return response_error("method")
+    elif not version_validation(request):
+        return response_error("version")
+    elif not host_validation(request):
+        return response_error("host")
+    elif not format_validation(request):
+        return response_error("format")
+    uri = request.split(" ", 2)[1]
+    info = resolve_uri(uri)
+    return response_error("OK", info[0], info[1])
+
+
+def method_validation(request):
+    """Verify method formatting and type."""
+    return request[:4] == u"GET "
+
+
+def version_validation(request):
+    """Verify version formatting and type."""
+    version = request.split(' ')[2][:8]
+    return version == u"HTTP/1.1"
+
+
+def host_validation(request):
+    """Verify host formatting and type."""
+    host = request.split('\r\n')[3]
+    return host == u"Host:  "
+
+
+def format_validation(request):
+    """Verify format formatting and type."""
+    val_count = 0
+    for ind in range(0, len(request)):
+        if val_count == 0 or val_count == 1:
+            if request[ind] == u" ":
+                val_count += 1
+            elif request[ind:ind + 1] == u"\r\n":
+                return False
+        elif val_count == 2:
+            if request[ind:ind + 2] == u"\r\n":
+                val_count += 1
+            elif request[ind] == u" ":
+                return False
+        elif val_count % 2 == 1:
+            if request[ind] == u":":
+                val_count += 1
+            elif request[ind] == u" ":
+                return False
+        else:
+            if request[ind] == u"\r\n":
+                val_count += 1
+    return True
+
+
+def resolve_uri(uri):
+    """Uri determines response body content."""
+    import io
+    import os
+    if "." not in uri:
```

Comment:
Not the best way to determine whether or not something is a file/directory. Some paths have periods in them while still being directories. And some files have no extensions